crates/bevy_ecs/macros/src/component.rs

```diff
@@ -109,6 +133,12 @@ fn parse_component_attr(ast: &DeriveInput) -> Result<Attrs> {
     Ok(attrs)
 }
 
+#[cfg(feature = "bevy_reflect")]
+fn has_reflect_attr(ast: &DeriveInput) -> bool {
+    const REFLECT: &str = "reflect";
+    ast.attrs.iter().any(|a| a.path().is_ident(REFLECT))
+}
```

Comment:
I think that this should only return true if it's a `reflect` attribute _that registers `ReflectComponent`_.

There could be cases where a user wants to use reflection but not want it to register `ReflectComponent` (maybe to exclude it from scenes by default).

Without this, the type would always be registered with no way to undo it.

Reply:
I did this primarily because I'm horribly unfamiliar with proc macros. I'll take a stab at doing this soon. What's the best way to check list of tokens in the `reflect` list?

Reply:
Hm, maybe something like this?

> [!warning]
> Untested

```rust
ast.attrs.iter().any(|attr| {
  if !attr.path().is_ident(REFLECT) {
    return false;
  }

  if !matches(attr.meta, Meta::List(_)) {
    return false;
  }

  let mut is_component_registered = false;

  attr.parse_nested_meta(|meta| {
    is_component_registered |= meta.path.is_ident("Component");
    Ok(())
  })
  .ok(); 

  is_component_registered
})
```

Reply:
It does seem like this works.